mypy/messages.py

```diff
@@ -460,11 +460,26 @@ def incompatible_argument(self, n: int, m: int, callee: CallableType, arg_type:
             target = 'to {} '.format(name)
 
         msg = ''
+        try:
+            expected_type = callee.arg_types[m - 1]
+        except IndexError:  # Varargs callees
```

Comment:
I only moved that existing code up; if it is an issue it could be fix in another PR/commit.